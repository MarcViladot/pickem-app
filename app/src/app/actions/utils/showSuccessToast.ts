import i18next from 'i18next';
import Toast from 'react-native-toast-message';

export const SHOW_SUCCESS_TOAST = 'SHOW_SUCCESS_TOAST';
export const showSuccessToast = (msg?: string) => {
  Toast.show({
    text1: msg || i18next.t('COMMON.OPERATION_OK'),
    type: 'success',
    position: 'bottom',
  });
  return {
    type: SHOW_SUCCESS_TOAST,
  };
};
