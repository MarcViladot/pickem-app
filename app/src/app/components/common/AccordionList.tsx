import React, { FC, useState } from "react";
import { StyleSheet, View, Text, Pressable } from "react-native";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { faChevronDown, faChevronUp } from "@fortawesome/free-solid-svg-icons";
import {useTheme} from '@react-navigation/native';
import {ThemeText} from './ThemeText';

interface Props {
  name: string;
  listItems: any[];
  openByDefault: boolean;
}

const AccordionList: FC<Props> = ({name, listItems, openByDefault}) => {

  const { colors } = useTheme();
  const [open, setOpen] = useState(openByDefault);

  return (
    <View>
      <View style={styles.listHeader}>
        <ThemeText style={styles.listName}>{name}</ThemeText>
        <Pressable onPress={() => setOpen(prevState => !prevState)} testID={"toggleButton"}>
          <FontAwesomeIcon icon={open ?  faChevronUp : faChevronDown} color={'gray'} size={14} />
        </Pressable>
      </View>
      { open &&
        <View style={styles.listContent} testID={"listContent"}>
          {listItems.map((item) => item)}
        </View>
      }
    </View>
  );
};

const styles = StyleSheet.create({
  listHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 10,
    paddingHorizontal: 12
  },
  listName: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  listContent: {
    paddingRight: 15,
    paddingLeft: 15
  }
})

export default AccordionList;
