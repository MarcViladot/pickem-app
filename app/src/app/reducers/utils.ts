// import {SHOW_LOADING} from '../actions/utils/showLoading';
// import {HIDE_LOADING} from '../actions/utils/hideLoading';
// import {SET_LANG} from '../actions/utils/setLang';
import { AnyAction } from "redux";

const initialState = {
    serverLoading: false,
    currentLang: null
};

export default (state = initialState, action: AnyAction) => {
    /*if (action.type === SHOW_LOADING) {
        return {
            ...state,
            serverLoading: true
        };
    } else if (action.type === HIDE_LOADING) {
        return {
            ...state,
            serverLoading: false
        };
    } else if (action.type === SET_LANG) {
        return {
            ...state,
            currentLang: action.lang
        };
    }*/
    return state;
};
