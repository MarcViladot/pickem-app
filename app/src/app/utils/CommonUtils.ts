import {Round} from '../interfaces/league.interface';

export class CommonUtils {

    public static getRoundName(round: Round, lang: string): string {
        const translatedName = round.translationGroup.roundNames.find((trans) => trans.lang === lang);
        if (translatedName) {
            return `${translatedName.text} ${round.translationNameExtra}`;
        }
        return round.name;
    }
}
