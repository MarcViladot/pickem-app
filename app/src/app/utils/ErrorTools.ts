import {ResponseApi, WebApiCodesTranslationMap, WebApiErrorCode} from './IResponse';


export class ErrorTools {

    public static getApiErrorMsg(errorCode: number, params: string[]) {
        const error = WebApiCodesTranslationMap.get(errorCode) || 'ERRORS.UNEXPECTED_ERROR';
        const paramsToTranslate = {};
        if (params) {
            params.forEach((param, i) => paramsToTranslate[`p${i + 1}`] = param);
        }
        return {error, paramsToTranslate};
    }

    static getApiErrorMsgFromResponse(res: ResponseApi<any>) {
        return ErrorTools.getApiErrorMsg(res.ErrorCode, res.ErrorParams);
    }

}
