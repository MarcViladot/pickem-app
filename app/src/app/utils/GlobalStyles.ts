import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    viewContainer: {
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
    },
    pageContainer: {
        padding: 5,
        flex: 1
    },
    pageContent: {
        width: '100%',
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#FFF',
        padding: 5,
        paddingTop: 0,
    },
    boldText: {
        fontWeight: 'bold'
    },
    row: {
        display: 'flex',
        flexDirection: 'row'
    },
    pageHeader: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        borderBottomWidth: 1,
        paddingBottom: 1,
        borderBottomColor: '#808080',
    },
    headerTitle: {
        fontSize: 22,
        fontFamily: 'Inter-Light'
    },
    dangerText: {
        color: '#BC1C61',
        textAlign: 'center',
        fontSize: 20,
    },
});
