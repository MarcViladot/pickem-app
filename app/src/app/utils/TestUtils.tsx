import store from '../../../store';
import {render} from '@testing-library/react-native'
import {Provider} from 'react-redux';
import React, {useState} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {DarkTheme, LightTheme, ThemeContext} from '../../../themes';
import {ThemeProvider} from 'styled-components/native';

const AllTheProviders = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={theme === 'dark' ? DarkTheme : LightTheme}>
                    {children}
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const ReduxProvider = ({children}) => {

    return (
        <Provider store={store}>
            {children}
        </Provider>
    )
};

const ProvidersWithNav = ({children}) => {
    const [theme, setTheme] = useState('dark');
    const themeData = {theme, setTheme};
    return (
        <Provider store={store}>
            <ThemeContext.Provider value={themeData}>
                <ThemeProvider theme={DarkTheme}>
                    <NavigationContainer theme={DarkTheme}>
                        {children}
                    </NavigationContainer>
                </ThemeProvider>
            </ThemeContext.Provider>
        </Provider>
    )
};

const customRender = (ui, options?) =>
    render(ui, {wrapper: AllTheProviders, ...options});

const reduxRender = (ui, options?) =>
    render(ui, {wrapper: ReduxProvider, ...options});

const navRender = (ui, options?) =>
    render(ui, {wrapper: ProvidersWithNav, ...options});

export * from '@testing-library/react-native'
export {customRender as render}
export {reduxRender as reduxRender}
export {navRender as navRender}
