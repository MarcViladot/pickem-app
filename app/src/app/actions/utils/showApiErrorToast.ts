import {ResponseApiError, WebApiErrorCode} from '../../utils/IResponse';
import {ErrorTools} from '../../utils/ErrorTools';
import Toast from 'react-native-toast-message';
import i18next from 'i18next';

export const SHOW_TOAST = "SHOW_TOAST";
export const showApiErrorToast = (res: ResponseApiError) => {
    const translation = ErrorTools.getApiErrorMsgFromResponse(res);
    Toast.show({
        text1: i18next.t(translation.error, translation.paramsToTranslate),
        type: res.ErrorCode !== -2 ? 'apiError' : 'serverError',
        position: 'bottom',
        props: res
    });
    return {
        type: SHOW_TOAST
    }
};
