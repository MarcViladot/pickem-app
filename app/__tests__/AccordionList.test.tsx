import {fireEvent, waitFor} from '@testing-library/react-native';
import {render} from '../src/app/utils/TestUtils';
import React from 'react';
import AccordionList from '../src/app/components/common/AccordionList';

describe('Accordion list component', () => {

    it('renders correctly', async () => {
        const {toJSON} = await waitFor(() =>
            render(<AccordionList name={"TestList"} listItems={[]} openByDefault={false}/>)
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render list title', async () => {
        const {queryByText} = await waitFor(() =>
            render(<AccordionList name={"TestList"} listItems={[]} openByDefault={false}/>)
        );
        const title = await queryByText("TestList");
        expect(title).toBeTruthy();
    });

    it('should not render list items if list is not open', async () => {
        const {queryByTestId} = await waitFor(() =>
            render(<AccordionList name={"TestList"} listItems={[]} openByDefault={false}/>)
        );
        const content = await queryByTestId("listContent");
        expect(content).toBeFalsy();
    });

    it('should render list items if list is open', async () => {
        const {queryByTestId} = await waitFor(() =>
            render(<AccordionList name={"TestList"} listItems={[]} openByDefault={false}/>)
        );
        const toggle = await queryByTestId("toggleButton");
        fireEvent.press(toggle);
        const content = await queryByTestId("listContent");
        expect(content).toBeTruthy();
    });

});
