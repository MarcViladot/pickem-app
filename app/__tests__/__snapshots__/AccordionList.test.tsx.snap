// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion list component renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
        "paddingLeft": 12,
        "paddingRight": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      TestList
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="toggleButton"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={14}
        bbWidth={14}
        fill="gray"
        focusable={false}
        height={14}
        meetOrSlice={0}
        minX={0}
        minY={0}
        secondaryFill={null}
        secondaryOpacity={0.4}
        size={14}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {},
            Object {
              "flex": 0,
              "height": 14,
              "width": 14,
            },
          ]
        }
        vbHeight={512}
        vbWidth={448}
        width={14}
      >
        <RNSVGGroup
          fill={4286611584}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill={4286611584}
            fillOpacity={1}
            propList={
              Array [
                "fill",
                "fillOpacity",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
